substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
+            if (level_Pointer.read() == NetinetIn.IPPROTO_IP()) {
+                //  1799                  return createBoolean(env, (int)!optval.c);
+                if (optval_c_Pointer.read() == 0) {
+                    return Boolean.TRUE;
+                }
+                return Boolean.FALSE;
+            } else {
+                //  1801                  return createBoolean(env, !optval.i);
+                if (optval_i_Pointer.read() == 0) {
+                    return Boolean.TRUE;
+                }
+                return Boolean.FALSE;
+            }
+        }
+        //  1804          case java_net_SocketOptions_SO_BROADCAST:
+        //  1805          case java_net_SocketOptions_SO_REUSEADDR:
+        else if (opt == SocketOptions.SO_BROADCAST || opt == SocketOptions.SO_REUSEADDR) {
+            //  1806              return createBoolean(env, optval.i);
+            if (optval_i_Pointer.read() == 0) {
+                return Boolean.TRUE;
+            }
+            return Boolean.FALSE;
+        }
+        //  1808          case java_net_SocketOptions_SO_SNDBUF:
+        //  1809          case java_net_SocketOptions_SO_RCVBUF:
+        //  1810          case java_net_SocketOptions_IP_TOS:
+        else if (opt == SocketOptions.SO_SNDBUF || opt == SocketOptions.SO_RCVBUF || opt == SocketOptions.IP_TOS) {
+            //  1811              return createInteger(env, optval.i);
+            return new Integer(optval_i_Pointer.read());
+        }
+        //  1815      /* should never reach here */
+        //  1816      return NULL;
+        return null;
     }
+    // @formatter:on
+
 
     @Substitute
     @SuppressWarnings({"static-method", "unused"})
     protected void socketSetOption0(int opt, Object val) {
         throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.socketSetOption0(int, Object)");
     }
 
+    /* Do not re-format commented out code: @formatter:off */
+    //    949  /*
+    //    950   * Class:     java_net_PlainDatagramSocketImpl
+    //    951   * Method:    datagramSocketCreate
+    //    952   * Signature: ()V
+    //    953   */
+    //    954  JNIEXPORT void JNICALL
+    //    955  Java_java_net_PlainDatagramSocketImpl_datagramSocketCreate(JNIEnv *env,
+    //    956                                                             jobject this)
     @Substitute
-    @SuppressWarnings({"static-method"})
-    protected void datagramSocketCreate() {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.datagramSocketCreate()");
+    @SuppressWarnings({"static-method", "unused"})
+    protected void datagramSocketCreate() throws SocketException {
+
+
+        //   957      jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
+        FileDescriptor fdObj = Util_java_net_DatagramSocketImpl.as_Target_java_net_DatagramSocketImpl(this).fd;
+        //   958      int arg, fd, t = 1;
+        int arg, fd, t = 1;
+
```

Comment:
You omitted 
```
959    char tmpbuf[1024];
```
You are not using it, but it would be good to leave the line in (commented out) with an explanation that you find Java Strings easier to deal with.  Otherwise readers get confused when they get to code that seems to use it.